### PR TITLE
Add exception for reinit, with example.

### PR DESCRIPTION
**Pull request by HansOlsson**
Based on decision in #3544

### CODE REVIEW

**Comment by HansOlsson** [comment]
Would be good to have it merged, as far as I understand it was based on the decision.

File: chapters/operatorsandexpressions.tex [hunk 1]
```diff
@@ -803,6 +804,20 @@ \subsection{Derivative and Special Purpose Operators with Function Syntax}\label
...
 
+\begin{example}
+Exception for reinit:
+\begin{lstlisting}[language=modelica]
+  Real x4, x5, x6;
+equation
+  when sample(1, 1) then
+    reinit(x4, 0);
+  end when;
+  der(x4) = time;
```
Comment (eshmoylova) [inline chapters/operatorsandexpressions.tex:815]: Is that allowed? It would be nice to add a comment about it here too.

**Review by eshmoylova — CHANGES_REQUESTED** [review]

File: chapters/operatorsandexpressions.tex (hunk 1 above)
Comment (HansOlsson) [reply chapters/operatorsandexpressions.tex:815]: I'm unsure what the question specifically refers to, but the entire example is intended to be legal.

**Review by eshmoylova — APPROVED** [review]

File: chapters/operatorsandexpressions.tex [hunk 2]
```diff
@@ -803,6 +804,20 @@ \subsection{Derivative and Special Purpose Operators with Function Syntax}\label
...
+  x5 = x4 * x4;
```
Comment (henrikt-ma) [inline chapters/operatorsandexpressions.tex:816]: Looks better:
```suggestion
  x5 = x4^2;
```

**Pull request merged**